packages/ses/src-xs/compartment-shim.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @module Provides a XS-specific variation on the behavior of
+ * ../compartment-shim.js, completing the story that begins in
+ * ./compartment.js, adding a Compartment constructor adapter to the global
+ * scope and transforming all of the methods of a native compartment into
+ * thunks that will alternately delegate to its native or shim behaviors
+ * depending on the __native__ Compartment constructor option.
+ */
+
+import { defineProperty, globalThis, weakmapGet } from '../src/commons.js';
+import {
+  NativeStartCompartment,
+  nativeCompartmentPrototype,
+  nativeImport,
+  nativeImportNow,
+  nativeEvaluate,
+  nativeGetGlobalThis,
+} from './commons.js';
+import {
+  ShimStartCompartment,
+  adaptCompartmentConstructors,
+  privateFields,
+  shimEvaluate,
+  shimGetGlobalThis,
+  shimImport,
+  shimImportNow,
+} from './compartment.js';
+
+const adapterFunctions = {
+  evaluate(source, options) {
+    const fields = weakmapGet(privateFields, this);
+    if (fields === undefined) {
+      return nativeEvaluate(source);
+    }
+    const { transforms, delegateNative, nativeEval } = fields;
+    for (const transform of transforms) {
+      source = transform(source);
+    }
```

Comment:
Do the imports from commons.js mean that this code needs to be robust against a tampered environment? If so, then it probably shouldn't use `of`.
```suggestion
    for (let i = 0; i < transforms.length; i += 1) {
      source = transforms[i](source);
    }
```